oauth_client.go

```diff
@@ -125,3 +126,40 @@ func (s *oAuthClients) Create(ctx context.Context, organization string, options
 
 	return oc, nil
 }
+
+type OAuthCLientDestroyOptions struct {
+	CLIENT_ID    *string `jsonapi:"primary,oauth-client"`
+	Organization *string `jsonapi:"primary,organization"`
+}
+
+func (o OAuthCLientDestroyOptions) valid() error {
+	if !validString(o.CLIENT_ID) {
+		return errors.New("CLIENT_ID required")
+	}
+	return nil
+}
+
+func (s *oAuthClients) Delete(ctx context.Context, options OAuthCLientDestroyOptions) (*OAuthClient, error) {
+	if err := options.valid(); err != nil {
+		return nil, err
+	}
+
+	u := fmt.Sprintf("oauth-clients/%s", url.QueryEscape(*options.CLIENT_ID))
+
+	req, err := s.client.newRequest("DELETE", u, s)
+	if err != nil {
+		return nil, err
+	}
+
+	oc := &OAuthClient{}
+	err = s.client.do(ctx, req, nil)
+	if err != nil && err.Error() != "resource not found" {
+		return nil, err
+	}
+
+	if err != nil && err.Error() == "resource not found" {
```

Comment:
I'm afraid I don't think I understand these error checks. Can you share your idea behind this?